Core/Services/StitchCoreLocalMongoDBService/Sources/StitchCoreLocalMongoDBService/LRUCache.swift

```diff
@@ -0,0 +1,75 @@
+internal final class LRUCache<Key: Hashable, Value>: Sequence {
+    typealias Element = (Key, Value)
+    typealias Iterator = LRUCacheIterator
+    fileprivate typealias Payload = (Key, Value, Int)
+    private var list = [Payload]()
```

Comment:
[minor] if we are sticking with an LRU cache, can we make this a dictionary for more efficient lookups? we can still enforce the same capacity restrictions

Reply:
Resolved offline.